### PR TITLE
Remove dependency on Spark internal Logging

### PR DESCRIPTION
**Pull request by jpdna**
Recent versions of Spark make Logging private, breaking logging code in ADAM and potentially other bdg projects.  This PR removes the dependency on `org.apache.spark.Logging` for projects using bdg-utils.

Note - Spark version is also bumped to 1.6.1 below, but this PR should work with Spark 1.5 as well

### CODE REVIEW

**Comment by heuermh** [comment]
Jenkins, test this please

File: utils-misc/src/main/scala/org/bdgenomics/utils/misc/LogUtils.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bdgenomics.utils.misc
+
+object LogUtils extends Logging {
```
Comment (heuermh) [inline utils-misc/src/main/scala/org/bdgenomics/utils/misc/LogUtils.scala]: I would move these methods to the private Logging object below

File: utils-misc/src/main/scala/org/bdgenomics/utils/misc/LogUtils.scala [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  def getADAMClassLoader: ClassLoader = getClass.getClassLoader
```
Comment (fnothaft) [inline utils-misc/src/main/scala/org/bdgenomics/utils/misc/LogUtils.scala]: `s/ADAM/BDG/g` or remove `ADAM`?

File: utils-misc/src/main/scala/org/bdgenomics/utils/misc/LogUtils.scala [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  def getContextOrADAMClassLoader: ClassLoader =
```
Comment (fnothaft) [inline utils-misc/src/main/scala/org/bdgenomics/utils/misc/LogUtils.scala]: Ditto RE: ADAM.

File: utils-misc/src/main/scala/org/bdgenomics/utils/misc/LogUtils.scala [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    Option(Thread.currentThread().getContextClassLoader).getOrElse(getADAMClassLoader)
+
+  def classForName(className: String): Class[_] = {
+    Class.forName(className, true, getContextOrADAMClassLoader)
+    // scalastyle:on classforname
```
Comment (fnothaft) [inline utils-misc/src/main/scala/org/bdgenomics/utils/misc/LogUtils.scala]: What's this pragma for?

File: utils-misc/src/main/scala/org/bdgenomics/utils/misc/LogUtils.scala (hunk 1 above)
Comment (fnothaft) [inline utils-misc/src/main/scala/org/bdgenomics/utils/misc/LogUtils.scala]: I think I agree. What is the reasoning for having these in a separate object?

File: pom.xml [hunk 5]
```diff
@@ -18,7 +18,7 @@
 
   <properties>
     <java.version>1.7</java.version>
-    <spark.version>1.5.2</spark.version>
+    <spark.version>1.6.1</spark.version>
```
Comment (fnothaft) [inline pom.xml]: Can you separate the Spark version bump into a separate commit?

Comment (fnothaft) [inline pom.xml]: And this will be nit picky, but can you make that commit _before_ the logging class changes?

File: utils-statistics/src/main/scala/org/bdgenomics/utils/statistics/normalization/ZScoreNormalization.scala [hunk 6]
```diff
@@ -17,7 +17,8 @@
  */
 package org.bdgenomics.utils.statistics.normalization
 
-import org.apache.spark.Logging
+import org.bdgenomics.utils.misc.Logging
+
```
Comment (fnothaft) [inline utils-statistics/src/main/scala/org/bdgenomics/utils/statistics/normalization/ZScoreNormalization.scala]: Nit: extra whitespace.

**Comment by fnothaft** [comment]
LGTM once nits are cleaned up. Thanks @jpdna!

**Comment by jpdna** [comment]
I think the above issues have been addressed with changes I pushed above, ending with 	6e5616e
ready for review again

File: utils-misc/src/main/scala/org/bdgenomics/utils/misc/Logging.scala [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bdgenomics.utils.misc
+
+//import org.apache.log4j.{ LogManager, PropertyConfigurator }
```
Comment (fnothaft) [inline utils-misc/src/main/scala/org/bdgenomics/utils/misc/Logging.scala]: Can you remove the dead includes?

**Comment by fnothaft** [comment]
LGTM other than the nit RE: dead includes. Ping me and I'll merge and cut a release—prob tomorrow PM.

**Comment by jpdna** [comment]
The dead imports are removed and squashed - let me know if anything else is needed to merge @fnothaft - thanks!

**Comment by fnothaft** [comment]
Thanks @jpdna—I will merge when Jenkins retests.

**Comment by heuermh** [comment]
LGTM

**Comment by fnothaft** [comment]
Jenkins, add to whitelist.

**Comment by fnothaft** [comment]
Jenkins, test this please.

**Comment by jpdna** [comment]
@fnothaft , just to double check, you included this PR in the upcoming release you are cutting correct?

**Comment by fnothaft** [comment]
LOL, oh god, I forgot this PR in the release I cut this morning. Oh man, that's a bit mortifying... I am blaming this on fatigue. I will cut a fixed release tomorrow AM. Sorry for the confusion. Let me merge this now, so that I do not mess this up tomorrow!
